arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Resource.kt

```diff
@@ -303,24 +301,21 @@ public interface ResourceScope : AutoCloseScope {
   public suspend fun <A> install(
     acquire: suspend AcquireStep.() -> A,
     release: suspend (A, ExitCase) -> Unit,
-  ): A
+  ): A = acquire(AcquireStep).also { a -> onRelease { release(a, it) } }
 
   /** Composes a [release] action to a [Resource] value before binding. */
   @ResourceDSL
-  public suspend infix fun <A> Resource<A>.release(release: suspend (A) -> Unit): A {
-    val a = bind()
-    return install({ a }) { aa, _ -> release(aa) }
-  }
+  public suspend infix fun <A> Resource<A>.release(release: suspend (A) -> Unit): A =
+    bind().also { a -> onRelease { release(a) } }
 
   /** Composes a [releaseCase] action to a [Resource] value before binding. */
   @ResourceDSL
-  public suspend infix fun <A> Resource<A>.releaseCase(release: suspend (A, ExitCase) -> Unit): A {
-    val a = bind()
-    return install({ a }, release)
-  }
+  public suspend infix fun <A> Resource<A>.releaseCase(release: suspend (A, ExitCase) -> Unit): A =
+    bind().also { a -> onRelease { release(a, it) } }
+
+  override fun onClose(release: (Throwable?) -> Unit): Unit = onRelease { release(it.errorOrNull) }
 
-  public suspend infix fun onRelease(release: suspend (ExitCase) -> Unit): Unit =
-    install({ }) { _, exitCase -> release(exitCase) }
+  public infix fun onRelease(release: suspend (ExitCase) -> Unit)
```

Comment:
not even marked as `suspend` anymore!

Reply:
I'm not entirely certain `infix` has much use here, given `ResourceScope` is predominantly used as a receiver. I can't see anyone wanting to write `this onRelease { ... }`. It _would_ impact source compatibility but ... is it ever actually used in that way?

Reply:
I agree, but I didn't wanna touch it. Maybe some folks who don't use contexts are forced to explicitly pass around `ResourceScope`, and hence they end up calling `scope onRelease { }`

Reply:
If this was `infix`, let's just keep it that way. No need to break source compatibility here.